online-rest-api/src/main/java/eu/itesla_project/online/rest/model/ViolationSynthesis.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2016, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package eu.itesla_project.online.rest.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+*
+* @author Giovanni Ferrari <giovanni.ferrari@techrain.it>
+*/
+public class ViolationSynthesis {
+    private final String equipment;
+    private final String type;
+    private final float limit;
+    private final List<TimeValue> timeValues = new ArrayList<TimeValue>();
+
+    public ViolationSynthesis(String equipment, String type, float limit ) {
+        Objects.requireNonNull(equipment);
+        Objects.requireNonNull(type);
+        this.equipment = equipment;
+        this.type = type;
```

Comment:
this.equipment = Objects.requireNonNull(equipment);
this.type = Objects.requireNonNull(type);

For simple assignment, you can use the shorter form.